### PR TITLE
lib/posix-futex/test: port the testsuite to the new sched apis

### LINKED ISSUE #630: posix-futex: Buiding tests fails on newer scheduling API
### Describe the bug

Changes in the scheduling API, as part of PR #564 cause the build process of [`posix-futex` tests](https://github.com/unikraft/unikraft/blob/staging/lib/posix-futex/tests/test_posix_futex.c) to fail. They are using the old style `uk_thread_create()` function. The newer API uses `uk_sched_thread_create()`.

### Steps to reproduce

Use the [`do-helloworld` Musl test script](https://github.com/unikraft-upb/scripts/tree/main/musl#app-helloworld). After running:

```
./do-helloworld build
```

enter the `workdir/apps/app-helloworld/` directory, run

```
export UK_WORKDIR=$(pwd)/../..
kraft menuconfig
````

and enable `uktest` and tests for `posix-futex`. Then get back to the top-level directory and run again

```
./do-helloworld build
```

This is when the build fails.

### Expected behavior

There shouldn't be a build / link error.

### Which architectures were you using or does this bug affect?

_No response_

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
/home/razvan/projects/unikraft/scripts/musl/workdir/apps/app-helloworld/build/helloworld_kvm-x86_64.o: In function `_uk_testsuite_posix_futex_testsuite_case_test_cmp_requeue_two_waite
rs_one_waker':                                                                                                                                                                         
/home/razvan/projects/unikraft/scripts/musl/workdir/unikraft/lib/posix-futex/tests/test_posix_futex.c:393: undefined reference to `uk_thread_create'                                   
/home/razvan/projects/unikraft/scripts/musl/workdir/unikraft/lib/posix-futex/tests/test_posix_futex.c:394: undefined reference to `uk_thread_create'                                   
/home/razvan/projects/unikraft/scripts/musl/workdir/unikraft/lib/posix-futex/tests/test_posix_futex.c:395: undefined reference to `uk_thread_create'                                   
/home/razvan/projects/unikraft/scripts/musl/workdir/unikraft/lib/posix-futex/tests/test_posix_futex.c:397: undefined reference to `uk_thread_wait'                                     
/home/razvan/projects/unikraft/scripts/musl/workdir/unikraft/lib/posix-futex/tests/test_posix_futex.c:402: undefined reference to `uk_thread_create'
```

### PR DESCRIPTION
**Pull request by noureddine-taleb**
GitHub-Fixes: #630

### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): x86_64
 - Platform(s): kvm
 - Lib(s): posix-futex

### Description of changes

Fix for https://github.com/unikraft/unikraft/issues/630

### CODE REVIEW

File: lib/posix-futex/tests/test_posix_futex.c [hunk 1]
```diff
@@ -109,7 +109,7 @@ static int futex(uint32_t *uaddr, int futex_op, uint32_t val,
  * Wait for the futex value to be equal to the iteration number and then
  * increment a variable.
```
Comment (Sairajkodilkar) [inline lib/posix-futex/tests/test_posix_futex.c:110]: Hi @noureddine-taleb,
Please include the description in your commit message,
This description generally includes overview of changes, need of changes and general background

File: lib/posix-futex/tests/test_posix_futex.c [hunk 2]
```diff
@@ -109,7 +109,7 @@ static int futex(uint32_t *uaddr, int futex_op, uint32_t val,
...
  */
```
Comment (Sairajkodilkar) [inline lib/posix-futex/tests/test_posix_futex.c:111]: The futex-tests depend on the syscall-shim library. Hence can you please add this dependancy in the Config.uk file.

**Review by Sairajkodilkar — CHANGES_REQUESTED** [review]

File: lib/posix-futex/tests/test_posix_futex.c (hunk 1 above)
Comment (noureddine-taleb) [reply lib/posix-futex/tests/test_posix_futex.c:110]: ok, I've added more description to the commit message.

File: lib/posix-futex/tests/test_posix_futex.c (hunk 2 above)
Comment (noureddine-taleb) [reply lib/posix-futex/tests/test_posix_futex.c:111]: done

**Comment by craciunoiuc** [comment]
I passed all tests, what do I win? 😎 

@adinasm @Sairajkodilkar any comments from your side? From my perspective, in this case, if the tests pass, it should be fine

**Review by adinasm — APPROVED** [review]
It's all good from my side. Thanks a lot!

Reviewed-by: Adina Smeu <adina.smeu@gmail.com>

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request closed without merging**